frontend/src/components/ui/custom/stories.tsx

```diff
@@ -0,0 +1,97 @@
+"use client";
+
+import * as React from "react";
+import { Avatar, AvatarFallback, AvatarImage } from "src/components/ui/avatar";
+import { cn } from "src/lib/utils";
+
+// Types for our Stories component
+export interface Story {
+  id: number;
+  avatar: string | React.ComponentType<{ className?: string }>;
+  title: string;
+  seen: boolean;
+  videoUrl?: string; // YouTube video URL
+}
+
+interface StoryItemProps extends React.HTMLAttributes<HTMLDivElement> {
+  story: Story;
+  onStoryClick?: (id: number) => void;
+}
+
+interface StoriesProps extends React.HTMLAttributes<HTMLDivElement> {
+  stories: Story[];
+  onStoryClick?: (id: number) => void;
+}
+
+// Individual story item component
+export function StoryItem({
+  story,
+  onStoryClick,
+  className,
+  ...props
+}: StoryItemProps) {
+  const handleClick = (e: React.MouseEvent<HTMLDivElement>) => {
+    if (onStoryClick) {
+      onStoryClick(story.id);
+    }
+  };
+
+  return (
+    <div
+      className={cn("flex flex-col items-center gap-1", className)}
+      {...props}
+    >
+      <div
+        className={cn(
+          "rounded-full border-2 flex items-center justify-center cursor-pointer p-0.5",
+          story.seen
+            ? "bg-muted" // Seen state
+            : "border-primary" // Unseen state with animated primary color border
+        )}
+        onClick={handleClick}
+      >
+        <Avatar className={cn("border-background size-14")}>
+          {typeof story.avatar === "string" ? (
```

Comment:
It would be good to just be consistent here and always use image URLs?

Reply:
Sure, it's currently in the state of "vibe-coded hack" :smile: